.pre-commit-config.yaml

```diff
@@ -35,3 +51,14 @@ repos:
       pass_filenames: false
       types: [text]
       verbose: true
+-   repo: https://github.com/ejba/pre-commit-maven
+    rev: v0.3.0
+    hooks:
+    - id: maven
+      args: ['tidy:pom', 'license:format']
+    - id: maven-checkstyle
+    - id: maven-spotless-apply
```

Comment:
should this rather be `maven-spotless-check`? Unless auto-formatting stuff is what we want

Reply:
I think it depends what we want. The current behaviour for hte python stuff is:

1) `git add file`
2) `git commit`
3) pre-commit fails because of black formatting. Black reformats file in staging
4) check staged changes and `git add file` again
5) `git commit`

So files that are modified aren't `add`ed and we can check the changes the formatters made. From that perspective I am happy w/ autoformatting. Especially for something like formatting a license, tidy-ing poms, spotless formatting: if we run `license:check` in pre-commit and it fails we will immediately run `license:format` and `git add`. 

What do you think?

Reply:
The goal of this is really to save us time w/ CI builds and catch dumb things before CI does.

Reply:
sgtm